### PR TITLE
Interaction Tracking

### PR DESCRIPTION
**Pull request by millerw4**
- added ClickTracker HOC and relevant React Context
- implemented ClickTracker in all Related Items Code
- added server-side logic to post interactions to HR server via provided API

### CODE REVIEW

**Review by tanyunzhe98 — APPROVED** [review]
looks good

File: src/Components/RelatedItems/RelatedItems.jsx [hunk 1]
```diff
@@ -147,7 +39,6 @@ var RelatedItems = ({current, setCurrentById, getProducts}) => {
         comparison.push([currentFeatures[combinedKeys[i]], combinedKeys[i], comparatorFeatures[combinedKeys[i]]])
       // }
```
Comment (Jlane20) [inline src/Components/RelatedItems/RelatedItems.jsx:40]: Two comparison variables in the same document. I though t you were assigning values directly to the STATE comparison which is something to not do and always use the "setting" variable. Further down I see you declared another comparison variable using var. 
Just something to maybe look out for. line 15 const comparison. and line 26 var comparison

**Review by Jlane20 — APPROVED** [review]
Looks good and thanks for the click tracking!

File: src/Components/RelatedItems/RelatedItems.jsx (hunk 1 above)
Comment (millerw4) [reply src/Components/RelatedItems/RelatedItems.jsx:40]: Good catch- the scoping works out okay in this case, but it does not read well at all. I'll make a note to modify this in the next update. Thanks!

**Pull request merged**